### PR TITLE
Make benchmark progress reports environment-aware

### PR DESCRIPTION
**Pull request by terceiro**
benchmarks.comparison has also been changed to use the stored data files
instead of recreating them from the DB.

### CODE REVIEW

File: benchmarks/progress.py [hunk 1]
```diff
@@ -67,3 +67,24 @@ def get_progress_since(date):
...
+            progress = Progress(
+                project=current.name,
+                branch=current.branch_name,
+                environment=environment,
+                before=baseline_testjob,
+                after=current_testjob,
+            )
+            result.append(progress)
+
+    return result
```
Comment (mwasilew) [inline benchmarks/progress.py]: What happens if there is no baseline? Will that just generate empty email?

Comment (terceiro) [inline benchmarks/progress.py]: This _if_ is here to handle the case where a build B has test jobs for environments E1, E2, ..., En, but the baseline build does not have test job results for all of those environments. In this case, the environments that are not covered in the baseline would be ignored in the comparison.

File: benchmarks/progress.py [hunk 2]
```diff
@@ -67,3 +67,24 @@ def get_progress_since(date):
                 result.append(progress)
 
     return result
+
+
+def get_progress_betwewn_results(current, baseline):
```
Comment (mwasilew) [inline benchmarks/progress.py]: there is a typo in the function name
currently: betwewn
probably should be: between

It is consistent in the call in tasks.py, so there is no problem right now :)

File: benchmarks/progress.py (hunk 1 above)
Comment (mwasilew) [inline benchmarks/progress.py]: ok, my mistake. The missing baseline case is handled elsewhere.

File: benchmarks/tasks.py [hunk 3]
```diff
@@ -325,28 +325,27 @@ def sync_external_repos(self):
 @celery_app.task(bind=True)
 def daily_benchmark_progress(self):
     now = timezone.now()
-    interval = relativedelta(days=1)
+    yesterday = now - relativedelta(days=1)
 
-    results = models.Result.objects.compare_progress(now, interval)
```
Comment (mwasilew) [inline benchmarks/tasks.py]: do we still need compare_progress function? It's left there in the ResultManager and the only place it's used is tests.

**Comment by mwasilew** [comment]
I think it's OK to be merged. The comments can be addressed later. I guess the next item is sandboxing the compare.py as guys requested some additional libraries to be used.

**Pull request merged**